src/charm.py

```diff
@@ -190,55 +170,53 @@ def ensure_state(self):
         )
 
     def handle_ingress(self):
-        ingress = self._get_interface("ingress")
-
-        if ingress:
-            for app_name, version in ingress.versions.items():
-                data = {
-                    "prefix": "/dex",
-                    "rewrite": "/dex",
-                    "service": self.model.app.name,
-                    "port": self.model.config["port"],
-                }
+        interface = self._get_interface("ingress")
+
+        if not interface:
+            return
+
+        data = {
+            "prefix": "/dex",
+            "rewrite": "/dex",
+            "service": self.model.app.name,
+            "port": self.model.config["port"],
+        }
 
-                ingress.send_data(data, app_name)
+        interface.send_data(data)
```

Comment:
Comparing this to the previous version, I'm not sure how this can work.  Looks like in the old code we looped through `interface.items()`, whereas now we just use it directly.  Did this get changed somewhere that I'm missing?

Reply:
The tests are passing so I think I'm missing something (or the tests are)

Reply:
it was looping over versions of schemas. I suppose new SDI does it in backend

Reply:
No I believe it was looping over relations, not versions of schema.  That's why we get a different `app_name` each time

Reply:
I don't think SDI is doing anything different on the backend here.  I think we 're just sending the same data to all related apps rather than data to each individually.  That should be fine though.